trinity/utils/chains.py

```diff
@@ -83,48 +83,48 @@ def get_nodekey_path(data_dir: PurePath) -> PurePath:
 DATABASE_SOCKET_FILENAME = 'db.ipc'
 
 
-def get_database_socket_path(data_dir: PurePath) -> str:
+def get_database_socket_path(data_dir: Path) -> str:
     """
     Returns the path to the private key used for devp2p connections.
 
     We're still returning 'str' here on ipc-related path because an issue with
-    multi-processing not being able to interpret 'PurePath' objects correctly.
+    multi-processing not being able to interpret 'Path' objects correctly.
     """
     return os.environ.get(
         'TRINITY_DATABASE_IPC',
-        os.path.join(str(data_dir), DATABASE_SOCKET_FILENAME),
+        str(data_dir / DATABASE_SOCKET_FILENAME),
```

Comment:
What I meant to say [here](https://github.com/ethereum/py-evm/pull/756#discussion_r190263967) was that the `/` looks wrong to me. To be honest, I didn't even know this syntax was possible because for simple strings this would result in an error:

```python
str('foo' / 'bar')
TypeError: unsupported operand type(s) for /: 'str' and 'str'
```

So, I just learned that with `Path` that syntax is in fact possible

```Python
str(Path('foo') / 'bar')
'foo/bar'
```

However, wouldn't something like `os.path.join(data_dir, 'foo')` still be clearer? If that's an accepted Python thing, I'm happy to keep it the way it is.

@pipermerriam @carver

Reply:
Yeah, I visually rejected it the first time I saw it as well.  A full read through the pathlib docs has me convinced this is the right way to go across the board with path manipulation.